projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -124,8 +125,7 @@ public function add_plugins_menu() {
 		 * @since 10.3
 		 * @param bool $wpcom_marketplace_enabled Load the WordPress.com Marketplace feature. Default to false.
 		 */
-		if ( apply_filters( 'wpcom_marketplace_enabled', false ) ) {
-			global $submenu;
+		if ( isset( $submenu['plugins.php'] ) && apply_filters( 'wpcom_marketplace_enabled', false ) ) {
```

Comment:
I think we may have to keep the docblock right before the filter, so the parser can keep it up? Something like this:

```diff
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php
index 4c18642de2..8597cadaa8 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php
@@ -115,17 +115,20 @@ class Atomic_Admin_Menu extends Admin_Menu {
 	 */
 	public function add_plugins_menu() {
 		global $submenu;
-		/**
-		 * Whether to enable the marketplace feature entrypoint.
-		 * This filter is specific to WPCOM, that's why there is no
-		 * need to use `jetpack_` prefix.
-		 *
-		 * @use add_filter( 'wpcom_marketplace_enabled', '__return_true' );
-		 * @module masterbar
-		 * @since 10.3
-		 * @param bool $wpcom_marketplace_enabled Load the WordPress.com Marketplace feature. Default to false.
-		 */
-		if ( isset( $submenu['plugins.php'] ) && apply_filters( 'wpcom_marketplace_enabled', false ) ) {
+		if (
+			isset( $submenu['plugins.php'] )
+			/**
+			 * Whether to enable the marketplace feature entrypoint.
+			 * This filter is specific to WPCOM, that's why there is no
+			 * need to use `jetpack_` prefix.
+			 *
+			 * @use add_filter( 'wpcom_marketplace_enabled', '__return_true' );
+			 * @module masterbar
+			 * @since 10.3
+			 * @param bool $wpcom_marketplace_enabled Load the WordPress.com Marketplace feature. Default to false.
+			 */
+			&& apply_filters( 'wpcom_marketplace_enabled', false )
+		) {
 			$plugins_submenu = $submenu['plugins.php'];
 			$slug_to_update  = 'plugin-install.php';
 

````

Reply:
c28a60a050